### ISSUE TITLE
Update 12.2.5 to 12.3.3 failed

### ISSUE THREAD

**Issue by FoxD7** [jboxberger/synology-gitlab#38]
Hi Juri,

my database migration failed during update. 

Log Entry: **Multiple indexes found on releases columns [:project_id]. Specify an index name from index_releases_on_project_id, releases_project_id_idx**

I found a "solution" on gitlab.com:
https://gitlab.com/gitlab-org/gitlab/issues/33084

> Fix
> Looking at `db/schema.rb` neither of the two `(project_id)` indexes is needed anymore. Running `DROP INDEX index_releases_on_project_id;` to drop one of the two `(project_id)` indexes makes the migration succeed and continue (successfully) with all remaining migrations.

Unfortunately, I do not know how I can apply the fix. Can you help me?

Kind regards,
FoxD7

**Comment by jboxberger** [comment]
Hello Daniel,

thank you very much for supporting my package and providing this fix here. Unfortunately the database migration script is provided by the [sameersbn/gitlab](https://hub.docker.com/r/sameersbn/gitlab/tags) package and this one is using the Gitlab Repository. So the fix has to be applied on the Gitlab side an will come automatically to my package. 

This is a Gitlab database Migration issue, a am only wrapping the docker conainer into a easy to use  synology package.

To mention the Problem i have made a Wiki entry: [https://github.com/jboxberger/synology-gitlab/wiki
](https://github.com/jboxberger/synology-gitlab/wiki)

Do you have any information wether the bug has already been fixed in Gitlab and on which version this happened?

Kind Regards

**Comment by FoxD7** [comment]
Hi Juri,

i tried the fix "manually drop index" (after i restored my 12.2.5).

The following steps were proceed:

- Open Docker (via Synology WebUI)
- Open "Details" for Container synology_gitlab_postgresql
- Create new bash window (see commands for login to db, drop index and logout)
```
root@synology_gitlab_postgresql:/var/lib/postgresql# sudo -u postgres psql -d gitlab                                                              
psql (10.4 (Ubuntu 10.4-2.pgdg18.04+1))                                                                                                           
Type "help" for help.                                                                                                                             
                                                                                                                                                  
gitlab=# drop index index_releases_on_project_id;                                                                                                  
DROP INDEX                                                                                                                                        
gitlab=# \q
```
- Finally i started the manual update for GitLab (with your 12.3.3 Package)

After a few minutes the GitLab update incl. db migration was successfully done. No Errors occured!!
And of course - GitLab Container is no longer shutting down :)


Looking forward to your next version!


Daniel

**Comment by crux153** [comment]
Faced similar issue when updating to 12.4.1, related to index `ci_trigger_requests_trigger_id_idx`.
There's already an issue on https://gitlab.com/gitlab-org/gitlab/issues/35184 and seems to be fixed in 12.4.2.

Manual steps to fix (from gitlab.com):
```sql
DROP INDEX fk_b8ec8b7245;
DROP INDEX ci_trigger_requests_trigger_id_idx;
```

**Comment by jboxberger** [comment]
@crux153 thanks a lot for the provided information, i've added this to the 12.4.1 release.

I also published the new 12.4.2, hopefully this solves the issue completely.
https://github.com/jboxberger/synology-gitlab/releases
